### PR TITLE
revision

### PR DESCRIPTION
**Pull request by yjlolo**
re https://github.com/yoyololicon/mdx-submissions21/pull/2#issuecomment-932733297
Thanks for inviting, pls go ahead editing as you see fit.

### CODE REVIEW

File: paper.md [hunk 1]
```diff
@@ -24,55 +24,74 @@ bibliography: paper.bib
...
-Danna-Sep is a combination of three different models: X-UMX, Demucs, and U-Net. In the following sub-sections, the modifications made in each model will be discussed.
+Danna-Sep is a combination of three different models: X-UMX, U-Net, and Demucs.
+We describe the enhancements made for each model in the following subsections.
 
 ## X-UMX
-X-UMX [@sawata20] is an improved version of UMX[@stoter19] in which it concatenates different hidden layers of UMX to share information among all target instruments. We trained it using the same time domain loss as the original X-UMX, but modified the frequency domain loss such that:
+X-UMX [@sawata20] improved upon UMX[@stoter19] by concatenating hidden layers of UMX to enable sharing information among all target instruments.
+We trained the model using the same time-domain loss as the original X-UMX, but modified the frequency-domain loss for $j$-th source as follows:
 
 $$\mathcal{L}_{MSE}^J = \sum_{j=1}^J\sum_{t,f}|Y_j(t, f) - \hat{Y}_j(t, f)|^2$$,
```
Comment (yoyolicoris) [inline paper.md]:
```suggestion
$$\mathcal{L}_{MSE}^J = \sum_{j=1}^J\sum_{t,f}|Y_j(t, f) - \hat{Y}_j(t, f)|^2$$
```

File: paper.md [hunk 2]
```diff
@@ -24,55 +24,74 @@ bibliography: paper.bib
...
+The training loss aggregates the L1-norm between estimated and ground-truth waveforms of the four sources.
+The model took approximately 10 days to train on a single RTX 3070 using mixed precision with a batch size of 16, and four steps of gradient accumulation.
 
-|         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) |
-|---------|:-----:|:----:|:-----:|:------:|:------------:|:---------:|
-| X-UMX   | 0.2   | 0.1  | 0     | 0.2    | TF | 136
+## Danna-Sep
+In order to obtain the final output of our framework, we calculated weighted average of individual outputs from the above-mentioned models.
+Experiments were conducted to search for optimal weighting. The optimal weights for each source, types of input domain (T for waveforms, TF for frequency masking), and the sizes of the models are given in the following table.
+|         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) | |---------|:-----:|:----:|:-----:|:------:|:------------:|:---------:| | X-UMX   | 0.2   | 0.1  | 0     | 0.2    | TF | 136
```
Comment (yoyolicoris) [inline paper.md]:
```suggestion
|         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) | 
|---------|:-----:|:----:|:-----:|:------:|:------------:|:---------:| 
| X-UMX   | 0.2   | 0.1  | 0     | 0.2    | TF | 136
```
broken table

File: paper.md [hunk 3]
```diff
@@ -24,55 +24,74 @@ bibliography: paper.bib
...
 | U-Net   | 0.2   | 0.17 | 0.5   | 0.4    | TF | 61
 | Demucs  | 0.6   | 0.73 | 0.5   | 0.4    | T | 733
 
 All models were trained on the training set of musdb18-hq [@musdb18-hq] using an Adam optmizier[@kingma2014adam]. 
 
 # Separation performances
-We evaluated our models in terms of Signal-to-Distortion Ratio (SDR) [@vincent2006performance] on musdb18 [@musdb18] using the *museval* toolbox [@fabian_robert_stoter_2019_3376621]. One iteration of MWF was used for X-UMX and U-Net, and we didn't apply the shift trick [@defossez2019music] for our Demucs model. The results of our models and the original X-UMX and Demucs are shown in the table below.
+We performed the benchmark using musdb18-hq [@musdb18] as the dataset for a fair comparison, and evaluated the models in terms of Signal-to-Distortion Ratio (SDR) [@vincent2006performance] derived from *museval* [@fabian_robert_stoter_2019_3376621].
```
Comment (yoyolicoris) [inline paper.md]: We trained Danna-Sep on musdb18-hq (high quality) but the benchmark was done on musdb18 (low quality) so the score can be compared to other works. This is different from X-UMX and Demucs, so I add one more line to state that we picked baselines that were also trained on high quality dataset.
I prefer the previous version.

File: paper.md [hunk 4]
```diff
@@ -24,55 +24,76 @@ bibliography: paper.bib
...
+This makes the total number of parameters comparable with the original Demucs.
+The training loss aggregates the L1-norm between estimated and ground-truth waveforms of the four sources.
+The model took approximately 10 days to train on a single RTX 3070 using mixed precision with a batch size of 16, and four steps of gradient accumulation.
 
-|         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) |
-|---------|:-----:|:----:|:-----:|:------:|:------------:|:---------:|
+## Danna-Sep
+In order to obtain the final output of our framework, we calculated weighted average of individual outputs from the above-mentioned models.
+Experiments were conducted to search for optimal weighting. The optimal weights for each source, types of input domain (T for waveforms, TF for frequency masking), and the sizes of the models are given in the following table.
+|         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) |
```
Comment (yoyolicoris) [inline paper.md:67]:
```suggestion

|         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) | 
```

File: paper.md (hunk 3 above)
Comment (yjlolo) [reply paper.md]: Thanks for clarifying; refer to [commit](https://github.com/yoyololicon/mdx-submissions21/pull/4/commits/09ada99f83b65e567bf7c99a9d85371155d76190) for the update.

File: paper.md [hunk 5]
```diff
@@ -24,55 +24,77 @@ bibliography: paper.bib
...
 # Method
 
-Danna-Sep is a combination of three different models: X-UMX, Demucs, and U-Net. In the following sub-sections, the modifications made in each model will be discussed.
+Danna-Sep is a combination of three different models: X-UMX, U-Net, and Demucs.
+We describe the enhancements made for each model in the following subsections.
 
 ## X-UMX
-X-UMX [@sawata20] is an improved version of UMX[@stoter19] in which it concatenates different hidden layers of UMX to share information among all target instruments. We trained it using the same time domain loss as the original X-UMX, but modified the frequency domain loss such that:
+X-UMX [@sawata20] improved upon UMX[@stoter19] by concatenating hidden layers of UMX to enable sharing information among all target instruments.
+We trained the model using the same time-domain loss as the original X-UMX, but modified the frequency-domain loss for $j$-th source as follows:
```
Comment (yoyolicoris) [inline paper.md]:
```suggestion
We trained the model using the same time-domain loss as the original X-UMX, but modified the frequency-domain loss for $J$ sources as follows:
```

File: paper.md [hunk 6]
```diff
@@ -24,55 +24,77 @@ bibliography: paper.bib
...
-
-## Demucs
-For Demucs [@defossez2019music], we chose the version with 48 hidden channels as our starting point. Then we replaced the decoder part of the network with four independent decoders, each corresponds to one source. These four decoders have the same architecture as the original decoder network while the hidden channel size was reduced to 24 so the total number of parameters is roughly the same. The training loss is a L1-norm between predicted waveforms and source-target waveforms, and it took around 10 days to train on a single RTX 3070 using mixed precision with a batch size of 16, and 4 steps of gradient accumulation.
+where $Y(t, f)$ and $\hat{Y}(t, f)$ are ground-truth and estimated time-frequency representations, respectively.
```
Comment (yoyolicoris) [inline paper.md]:
```suggestion
where $Y_j(t, f)$ and $\hat{Y}_j(t, f)$ are ground-truth and estimated time-frequency representations for the $j$-th source, respectively. 
```

**Review by yoyolicoris — COMMENTED** [review]
Some minor suggestions around the equation part, otherwise LGTM!

**Comment by yoyolicoris** [comment]
Thanks for the help!

**Pull request merged**